### PR TITLE
Add `ccat` formula

### PR DESCRIPTION
**Pull request by jingweno**
Project: https://github.com/jingweno/ccat

### CODE REVIEW

File: Library/Formula/ccat.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require "formula"
```
Comment (DomT4) [inline Library/Formula/ccat.rb]: This hasn't been required for a while. Can :scissors: it.

File: Library/Formula/ccat.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+class Ccat < Formula
+  homepage "https://github.com/jingweno/ccat"
+  url "https://github.com/jingweno/ccat/archive/v0.1.0.tar.gz"
+  sha1 "fc0a1a49ca81fc85b9d50f48d63182d507b4627c"
```
Comment (DomT4) [inline Library/Formula/ccat.rb]: This should be `sha256`.

File: Library/Formula/ccat.rb (hunk 1 above)
Comment (jingweno) [inline Library/Formula/ccat.rb]: Done (`--amend`)

File: Library/Formula/ccat.rb (hunk 2 above)
Comment (jingweno) [inline Library/Formula/ccat.rb]: Didn't know we're using `sha256` forward. How about existing formulas? Any plan to fix all of them?

Comment (jingweno) [inline Library/Formula/ccat.rb]: Fixed (`--amend`)

**Comment by jingweno** [comment]
@DomT4 BTW, I'd want the build bot to add `bottle` as well. I assume I don't need to do anything?

File: Library/Formula/ccat.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/ccat.rb]: Yes. `sha256` is required for all new formulae, all existing `sha1` bottles are being migrated to `sha256` as they are updated and there's a strong ongoing effort to eventually deprecate `sha1` from formulae entirely.

**Comment by DomT4** [comment]
> I assume I don't need to do anything?

Nope. Bot generates bottles automatically.

File: Library/Formula/ccat.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  url "https://github.com/jingweno/ccat/archive/v0.1.0.tar.gz"
+  sha256 "437c7582ab39c8a1fb5e97ae9569e0096c98209a2ebb11ca67a7f9246b129c7c"
+
+  depends_on "go" => :build
+
+  def install
+    system "./script/build"
+    bin.install "ccat"
+  end
+end
```
Comment (DomT4) [inline Library/Formula/ccat.rb]: If there's anything you can do in regards to a test, that'd be great. Something like:

```ruby
(testpath/"test.txt").write <<-EOS.undent
  I am a colourful cat
EOS

assert_match /I am a colourful cat/, shell_output("#{bin}/ccat test.txt")
```

Should work. Test it first though with `brew test --verbose`, don't just take my word for it :wink:.

Comment (jingweno) [inline Library/Formula/ccat.rb]: Yes, that works. I added it (`--amend`). `ccat` is running in non terminal mode in the test so it falls back to work like `cat`.

**Comment by jingweno** [comment]
Build is green. Ready to be merged :)

**Comment by chdiza** [comment]
This is self-submitted, and at version 0.1.

**Comment by DomT4** [comment]
It also has 941 Github stars, which is usually enough for the maintainers, which is why I didn't bother pointing it out earlier. Indifferent though, I'll defer to those with the power to make the decision.

**Comment by jingweno** [comment]
@chdiza why would version 0.1 matter for a formula?

**Comment by chdiza** [comment]
> why would version 0.1 matter for a formula?

[Because](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md#niche-or-self-submitted-stuff) the docs say only stable packages are welcome.  A version number that low indicates that it's something very new.

I might also add, for the maintainers, that the whole project is six days old.

**Comment by jingweno** [comment]
> Because the docs say only stable packages are welcome. A version number that low indicates that it's something very new.

`ccat` is pretty stable - just look at the activity of the repo and what people said on Twitter. The reason I didn't mark it as 1.0 is because it doesn't have all features I want. If 1.0 is what people would expect a stable release before starting to use a tool, I could bump it to 1.0 right now. What I'm trying to say is, `ccat` is stable and people have already been using it.

**Comment by narqo** [comment]
:+1: for the merge

**Comment by mikemcquaid** [comment]
This is popular enough that I think we should merge it.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `ccat 0.1.0 (new formula)`.

**Comment by jingweno** [comment]
@mikemcquaid :beers:

**Pull request closed without merging**